tfjs-data/src/readers_test.ts

```diff
@@ -134,66 +136,62 @@ describeAllEnvs('readers', () => {
   });
 });
 
-describeBrowserEnvs('readers in browser', () => {
-  if (navigator.mediaDevices != null) {
-    it('generate data from webcam with HTML element', async () => {
-      setupFakeVideoStream();
-      const videoElement = document.createElement('video');
-      videoElement.width = 160;
-      videoElement.height = 90;
+describeWithFlags('readers in browser', MEDIA_ENVS, () => {
```

Comment:
MEDIA_ENVS is a new constraint set that only runs if navigator.mediaDevices is not null. It's required because jasmine 4 prohibits empty `describe` blocks, which the `if` statement would cause.